### PR TITLE
Account.childrenAccounts resolver

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/4383

### CODE REVIEW

File: server/graphql/v2/interface/Account.js [hunk 1]
```diff
@@ -568,6 +569,43 @@ const accountFieldsDefinition = () => ({
...
+      limit: { type: GraphQLInt, defaultValue: 100 },
+      offset: { type: GraphQLInt, defaultValue: 0 },
+      accountType: {
+        type: new GraphQLList(AccountType),
+      },
+    },
+    async resolve(account, args, req) {
+      if (!req.remoteUser?.isAdminOfCollective(account)) {
+        throw new Unauthorized('You need to be logged in as an admin of the account to see its children accounts');
+      }
```
Comment (znarf) [inline server/graphql/v2/interface/Account.js:584]: Why did you go for this? Doesn't seem necessary.

File: server/graphql/v2/interface/Account.js [hunk 2]
```diff
@@ -185,10 +185,11 @@ const accountFieldsDefinition = () => ({
...
 
-      let conditionOnMemberCollective;
-      if (args.type) {
-        const types = args.type.split(',');
-        conditionOnMemberCollective = { type: { [Op.in]: types } };
+      const conditionOnMemberCollective = {};
+      if (args.accountType && args.accountType.length > 0) {
+        conditionOnMemberCollective.type = {
+          [Op.in]: args.accountType.map(value => AccountTypeToModelMapping[value]),
+        };
```
Comment (znarf) [inline server/graphql/v2/interface/Account.js:192]: That's right. Useful for for Individual -> User.

Comment (kewitz) [reply server/graphql/v2/interface/Account.js:192]: Keeping it consistent with the existing behaviour for V2 account querying.

File: server/graphql/v2/interface/Account.js (hunk 1 above)
Comment (kewitz) [reply server/graphql/v2/interface/Account.js:584]: I'll remove this.

**Comment by znarf** [comment]
That was merged a bit fast. I wish Ben would have had the opportunity to look at this.

**Comment by kewitz** [comment]
@Betree, could you take a look at this PR? I underrated this issue a bit and merged it last week after François' review.
The main discussion I had with François after was related to if we should have this abstract children resolver or implement "events" and "projects" resolvers.

File: server/graphql/v2/interface/Account.js [hunk 3]
```diff
@@ -568,6 +569,39 @@ const accountFieldsDefinition = () => ({
       };
     },
   },
+  childrenAccounts: {
+    type: AccountCollection,
```
Comment (Betree) [inline server/graphql/v2/interface/Account.js:573]: We can make this non-nullable

File: server/graphql/v2/interface/Account.js [hunk 4]
```diff
@@ -568,6 +569,39 @@ const accountFieldsDefinition = () => ({
...
+    args: {
+      limit: { type: GraphQLInt, defaultValue: 100 },
+      offset: { type: GraphQLInt, defaultValue: 0 },
```
Comment (Betree) [inline server/graphql/v2/interface/Account.js:576]: These ones too, since they have a defaultValue. It's good practice to enforce a `limit`

**Review by Betree — COMMENTED** [review]
This looks great, I like the fact that we're implementing this as a generic field. 

Some minor feedback below but nothing blocking.

**Pull request merged**